File: solutions/go/rna-transcription/1/rna_transcription.go
package strand

var dnaToRnaMap = map[rune]string{
	'A': "U",
	'C': "G",
	'G': "C",
	'T': "A",
}

func ToRNA(dna string) (rna string) {
	for _, v := range dna {
		rna += dnaToRnaMap[v]
	}

	return
}

File: solutions/ruby/rna-transcription/1/rna_transcription.rb
# frozen_string_literal: true

# Handle DNA to RNA transcriptions
module Complement
  module_function

  DNA_TO_RNA = {
    'A' => 'U',
    'C' => 'G',
    'G' => 'C',
    'T' => 'A'
  }.freeze

  # @param rna [String]
  def of_dna(rna)
    rna.chars.map { |strand| DNA_TO_RNA[strand] }.join
  end
end
